metomi/rose/config_processors/fileinstall.py

```diff
@@ -891,12 +894,35 @@ def parse(self, loc, conf_tree):
             if handler is None:
                 raise ValueError(f"don't support scheme {loc.scheme}")
         else:
-            # Scheme not specified in the configuration.
+            # Try to get the scheme by parsing loc name, e.g. git:some-url
             scheme = urlparse(loc.name).scheme
             if scheme:
                 handler = self.get_handler(scheme)
                 if handler is None:
-                    handler = self.guess_handler(loc)
+                    # Try to guess the scheme using the ``can_handle`` method
+                    # from each handler in turn:
+                    try:
+                        handler = self.guess_handler(loc)
+
+                    # The rsync handler will attempt to run
+                    # ssh host test -e path
+                    # This logic allows us to report failure nicely.
+                    except PreRsyncCheck255Error as exc:
+                        raise ValueError(
+                            exc.BASE_MESSAGE
+                            + 'If it is then host'
+                            f' \"{loc.name.split(":")[0]}\"'
```

Comment:
Unnecessary escapes?
```suggestion
                            f' "{loc.name.split(":")[0]}"'
```